File: Stage18/Python/Problem25192.py
import sys
input = sys.stdin.readline

def main():
    N = int(input())
    group = set()
    count = 0

    for _ in range(N):
        name = input().strip()
        if name == "ENTER":
            group.clear()
        else:
            if name not in group:
                group.add(name)
                count += 1

    print(count)

if __name__ == "__main__":
    main()

File: Stage18/Java/Problem25192.java
package Stage18.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;
import java.util.Set;

public class Problem25192 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        Set<String> group = new HashSet<>();
        int count = 0;

        for (int i = 0; i < N; i++) {
            String name = br.readLine();

            if (name.equals("ENTER")) {
                group.clear();
            } else {
                if (!group.contains(name)) {
                    group.add(name);
                    count++;
                }
            }
        }

        System.out.println(count);
        br.close();
    }
}
